src/app/shared/picker/icon-picker/icon-picker.component.ts

```diff
@@ -143,7 +123,9 @@ export class IconPickerComponent implements OnInit, AfterViewInit {
   }
 
   public filterInput($event: KeyboardEvent): void {
-    this.filter = (<HTMLInputElement>$event.target).value;
+    if ($event) {
+      this.filter = (<HTMLInputElement>$event.target).value;
```

Comment:
This is just a detail but I have seen it a lot in your code. Angular binds event output values to `$event` variable in templates. However, when it comes to TypeScript code, the convention is to name variables in `camelCase` as you would do in Java. And it is recommended by some people to end such names with `$` if it is an observable so you know right away what you can do with it (especially if it is a class variable which you use in a template).